### PR TITLE
[Merged by Bors] - chore(Logic): remove bare open Classical

### PR DESCRIPTION
**Pull request by grunweg**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Logic/Nontrivial/Defs.lean [hunk 1]
```diff
@@ -44,6 +41,7 @@ protected theorem Decidable.exists_ne [Nontrivial α] [DecidableEq α] (x : α)
     exact ⟨y', h.symm⟩
   · exact ⟨y, Ne.symm hx⟩
 
+open Classical in
 theorem exists_ne [Nontrivial α] (x : α) : ∃ y, y ≠ x := Decidable.exists_ne x
```
Comment (grunweg) [inline Mathlib/Logic/Nontrivial/Defs.lean:45]: This is the only change I'm not fully sure of.

Comment (jcommelin) [reply Mathlib/Logic/Nontrivial/Defs.lean:45]: Why?

File: Mathlib/Logic/Denumerable.lean [hunk 2]
```diff
@@ -189,10 +189,9 @@ variable {s : Set ℕ} [Infinite s]
 
 section Classical
 
-open scoped Classical
-
-theorem exists_succ (x : s) : ∃ n, (x : ℕ) + n + 1 ∈ s :=
-  _root_.by_contradiction fun h =>
+theorem exists_succ (x : s) : ∃ n, (x : ℕ) + n + 1 ∈ s := by
+  classical
+  exact _root_.by_contradiction fun h =>
```
Comment (jcommelin) [inline Mathlib/Logic/Denumerable.lean:194]: Can this be optimized using `by_contra h` or something like that?

File: Mathlib/Logic/Nontrivial/Defs.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Logic/Nontrivial/Defs.lean:45]: I'm unsure in the sense of "should I add a suitable decidability assumption instead?" --- though come to think of it, the answer is "no".

File: Mathlib/Logic/Denumerable.lean (hunk 2 above)
Comment (grunweg) [reply Mathlib/Logic/Denumerable.lean]: Yes, it can - pushed.

**Review by j-loreaux — COMMENTED** [review]
Thanks!

bors merge

**Pull request closed without merging**